packages/@react-spectrum/color/src/HexColorField.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import React, {RefObject, useRef} from 'react';
+import {SpectrumHexColorFieldProps} from '@react-types/color';
+import {TextFieldBase} from '@react-spectrum/textfield';
+import {TextFieldRef} from '@react-types/textfield';
+import {useHexColorField} from '@react-aria/color';
+import {useHexColorFieldState} from '@react-stately/color';
+import {useProviderProps} from '@react-spectrum/provider';
+
+function HexColorField(props: SpectrumHexColorFieldProps, ref: RefObject<TextFieldRef>) {
+  props = useProviderProps(props);
+  let {
+    value,          // eslint-disable-line @typescript-eslint/no-unused-vars
```

Comment:
I'm going to leave this one as unresolved, I still don't like the solution, but can't think of something better yet. Maybe one of my teammates will have an idea, so I want them to see this.

Reply:
Hi @LFDanLu, I'm not quite sure how to follow your suggestion to resolve this.

It might be fine in SearchField's case because `defaultValue` is just a string but HexColorField's `value`, `defaultValue`, and `onChange` are not string types

```ts
export interface HexColorFieldProps {
  value?: ColorInput,
  defaultValue?: ColorInput,
  onChange?: (color: Color) => void
}
```

We have to omit these props before they're used in `useHexColorField`, otherwise hooks in it like `useTextField` will throw errors because it expects these props to be `string` type only.

Reply:
Ahh, my mistake, misunderstood the issue here. Hm will have to play around, but what you have might be the best solution.